react-app/src/components/ProposalDetailScreen/ProposalDetailScreen.tsx

```diff
@@ -1,21 +1,41 @@
 import React from "react";
 import { Navigate, useParams } from "react-router-dom";
-import cn from "classnames";
+import {
+  Proposal,
+  ProposalStatus,
+  ProposalType,
+} from "../../generated/graphql";
 import AppRoutes from "../../navigation/AppRoutes";
 import Paper from "../common/Paper/Paper";
+import ProposalHeader from "./ProposalHeader";
 
 const ProposalDetailScreen: React.FC = () => {
   const { id } = useParams();
 
   if (!id) return <Navigate to={AppRoutes.Proposals} />;
 
+  // mock data, remove after API integration
+  const proposal: Proposal = {
+    id,
+    description: "description",
+    type: ProposalType.Text,
+    proposerAddress: "like1ewpwcdfgsdfdu0jj2unwhjjl58yshm9xnvr9c2",
+    status: ProposalStatus.VotingPeriod,
+    title:
+      "Community-pool-spend Proposal: deposit 2,000,000 LIKE to Tech Subdao multisig wallet",
+    depositEndTime: new Date(),
+    submitTime: new Date(),
+    votingEndTime: new Date(),
+    votingStartTime: new Date(),
```

Comment:
note that depositEndTime, votingEndTime, votingStartTime might be null depending on the proposal status.

Reply:
Under what status would they be null?

Reply:
https://github.com/cosmos/cosmos-sdk/blob/7c4e4c1a44f3770c10956222b14b1f1595424172/x/gov/keeper/proposal.go#L237-L248
https://github.com/cosmos/cosmos-sdk/blob/7c4e4c1a44f3770c10956222b14b1f1595424172/x/gov/types/v1beta1/proposal.go#L31-L39

i think votingEndTime and votingStartTime would be null when proposalStatus == DepositPeriod, and depositEndTime becomes null when proposalStatus == VotingPeriod